pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,409 @@
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Requests.Cpu().Value()).To(BeNumerically("==", 3))
+			Expect(podResources.Requests.Memory().Value()).To(BeNumerically("==", 3221225472))
+			Expect(podResources.Limits.Cpu().Value()).To(BeNumerically("==", 6))
+			Expect(podResources.Limits.Memory().Value()).To(BeNumerically("==", 3221225472))
+		})
+
+		It("should schedule based on the sum of containers , sidecarContainers and overhead with max of initcontainers", func() {
+			pod := test.Pod(test.PodOptions{
+				Overhead: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("5"),
+					v1.ResourceMemory: resource.MustParse("500Mi"),
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("3"), v1.ResourceMemory: resource.MustParse("3Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("3Gi")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Requests.Cpu().Value()).To(BeNumerically("==", 8))
+			Expect(podResources.Requests.Memory().Value()).To(BeNumerically("==", 4819255296))
+			Expect(podResources.Limits.Cpu().Value()).To(BeNumerically("==", 10))
+			Expect(podResources.Limits.Memory().Value()).To(BeNumerically("==", 4819255296))
+		})
+
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers when there is a large initcontainer", func() {
```

Comment:
```suggestion
		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers when there is a large initContainer after a sidecarContainer", func() {
```

Reply:
Then you can create another test case: "should calculate resource requests when there is a large initContainer after a sidecarContainer that doesn't exceed container resource requests"